db/version_set.cc

```diff
@@ -4881,59 +4881,112 @@ Status VersionSet::WriteSnapshot(log::Writer* log) {
 uint64_t VersionSet::ApproximateSize(Version* v, const Slice& start,
                                      const Slice& end, int start_level,
                                      int end_level, TableReaderCaller caller) {
+  const auto& icmp = v->cfd_->internal_comparator();
+
   // pre-condition
-  assert(v->cfd_->internal_comparator().Compare(start, end) <= 0);
+  assert(icmp.Compare(start, end) <= 0);
 
   uint64_t size = 0;
   const auto* vstorage = v->storage_info();
-  end_level = end_level == -1
-                  ? vstorage->num_non_empty_levels()
-                  : std::min(end_level, vstorage->num_non_empty_levels());
+  const int num_non_empty_levels = vstorage->num_non_empty_levels();
+  end_level = (end_level == -1) ? num_non_empty_levels
+                                : std::min(end_level, num_non_empty_levels);
 
   assert(start_level <= end_level);
 
-  for (int level = start_level; level < end_level; level++) {
+  std::vector<FdWithKeyRange*> first_files;
```

Comment:
Might not a big deal in this case, but we tend to use our `autovector` in cases like this so that we can avoid malloc in some cases. We need to give a good enough value to allocate from the stack.

Reply:
is kSize == 16 good estimate?

Reply:
16 is a pretty good estimation.